core/file.cpp

```diff
@@ -267,6 +267,44 @@ bool remote_repo_uptodate(const char *filename, struct git_info *info)
 	return false;
 }
 
+static std::unique_ptr<std::vector<unsigned char>> read_into_buffer(const char *file)
+{
+	const char *failed_to_read_msg = translate("gettextFromC", "Failed to read '%s'");
+
+	struct stat file_status;
+	if (stat(file, &file_status) < 0) {
+		report_error(failed_to_read_msg, file);
+
+		return NULL;
+	}
+
+	// Open the archive
```

Comment:
archive feels just wrong here. "log" or why not just file, fd, f or something less confusing?